flaskext/sqlalchemy.py

```diff
@@ -620,6 +620,8 @@ def create_scoped_session(self, options=None, scopefunc=None):
         """
         if options is None:
             options = {}
+        if scopefunc is None:
+            scopefunc = _request_ctx_stack.__ident_func__
```

Comment:
This function does not return ident of current request (or current app in latest code base), instead it returns the ident of current thread. (this function is the built-in `_thread.get_ident` by default)
@davidism See https://github.com/pallets/werkzeug/blob/b3637ff4039200cc60c1a65ade422127127722a2/werkzeug/local.py#L121